src/dao/github/repository.js

```diff
@@ -0,0 +1,29 @@
+const BaseDao = require('~/src/dao').BaseDao
+const Repository = require('~/src/models/github/Repository')
+
+let dao
+
+exports.createDao = function () {
+  dao = new BaseDao({
+    modelType: Repository
+  })
+}
+
+/**
+* Upsert a new Repository record in the database, do nothing on conflict
+* @param id {Number} - github repository id
+* @param name {String} - name of repository
+* @param full_name {String} - owner + repository name e.g. druotic/cool-repo
+* @param installation_id {Number} - foreign key id for associated installation
+*/
+exports.upsert = async function ({ id, name, full_name, installation_id }) {
```

Comment:
`FashionKnex.prototype.upsert` already allows you to pass a `Model`. It might be cleaner to just do that. You may have to change up the way the `id` is passed though.

Reply:
For consistency, I was merely following your existing pattern in hooks of providing a passthrough for everything. This suggestion seems like a good candidate for refactoring PRs across the projects to help maintain consistency.